### PR TITLE
G0 5039 | New Datepicker

### PR DESCRIPTION
**Pull request by ravinderwalkover**
G0 5039 | New Datepicker

### CODE REVIEW

File: apps/web-giddh/src/app/contact/contact.component.html [hunk 1]
```diff
@@ -439,31 +445,31 @@ <h1 class="mrB1">Type message body:</h1>
                     <td></td>
                     <td *ngIf="showFieldFilter.parentGroup"></td>
                     <td [attr.colspan]="tableColsPan" class="text-center top-aging-table-top">
-                        <p style="display: flex;justify-content: center;align-items: center">
+                        <p style="display: flex;justify-content: center;align-items: center" container="body">
```
Comment (sadik-walkover) [inline apps/web-giddh/src/app/contact/contact.component.html:448]: Remove inline CSS

File: apps/web-giddh/src/app/contact/contact.component.html [hunk 2]
```diff
@@ -723,15 +729,15 @@ <h1 class="mrB1" style="margin-top:15px;">Type message body:</h1>
                         </td>
                         <td class="WordWrap " data-title="Parent Group" *ngIf="showFieldFilter.parentGroup">
                             {{account.groupUniqueName}}</td>
-                        <td [ngClass]="{'WordWrap text-right': 1 === 1, 'negative-balance': account.openingBalance?.type === 'CREDIT' && account.openingBalance?.amount }" data-title="Opening" *ngIf="showFieldFilter.openingBalance">
-                            <span *ngIf="account.openingBalance?.type === 'CREDIT' && account.openingBalance?.amount ">-</span>{{account.openingBalance?.amount |giddhCurrency}}
+                        <td class="WordWrap text-right  " data-title="Opening" *ngIf="showFieldFilter.openingBalance">
```
Comment (sadik-walkover) [inline apps/web-giddh/src/app/contact/contact.component.html:732]: change this class name WordWrap

File: apps/web-giddh/src/app/contact/contact.component.html [hunk 3]
```diff
@@ -723,15 +729,15 @@ <h1 class="mrB1" style="margin-top:15px;">Type message body:</h1>
...
+                            {{account.openingBalance?.amount |giddhCurrency}}
                         </td>
                         <td class="WordWrap text-right " data-title=" Receipts">
                             {{account.debitTotal |giddhCurrency}}</td>
                         <td class="WordWrap text-right " data-title=" Sales">
                             {{account.creditTotal |giddhCurrency}}</td>
-                        <td [ngClass]="{'WordWrap text-right': 1 === 1, 'negative-balance': account.closingBalance?.type === 'CREDIT' && account.closingBalance?.amount }" data-title="Closing">
-                            <span *ngIf="account.closingBalance?.type === 'CREDIT' && account.closingBalance?.amount ">-</span>{{account.closingBalance?.amount |giddhCurrency}}
+                        <td class="WordWrap text-right " data-title="Closing">
```
Comment (sadik-walkover) [inline apps/web-giddh/src/app/contact/contact.component.html:739]: change this class name WordWrap

File: apps/web-giddh/src/app/contact/contact.component.html [hunk 4]
```diff
@@ -924,16 +930,18 @@ <h1 class="mrB1" style="margin-top:15px;">Type message body:</h1>
                         <td class="WordWrap" data-title="Parent Group" *ngIf="showFieldFilter.parentGroup">
```
Comment (sadik-walkover) [inline apps/web-giddh/src/app/contact/contact.component.html:930]: change this class name WordWrap

File: apps/web-giddh/src/app/contact/contact.component.html [hunk 5]
```diff
@@ -924,16 +930,18 @@ <h1 class="mrB1" style="margin-top:15px;">Type message body:</h1>
...
                         </td>
                         <td class="WordWrap text-right " data-title="Purchase">
                             {{account.creditTotal |giddhCurrency}}</td>
 
                         <td class="WordWrap text-right " data-title="Payment">
                             {{account.debitTotal |giddhCurrency}}</td>
-                        <td [ngClass]="{'WordWrap text-right': 1 === 1, 'negative-balance': account.closingBalance?.type === 'DEBIT' && account.closingBalance?.amount }" data-title="Closing">
-                            <span *ngIf="account.closingBalance?.type === 'DEBIT' && account.closingBalance?.amount ">-</span>{{account.closingBalance?.amount |giddhCurrency}}
+
+
```
Comment (sadik-walkover) [inline apps/web-giddh/src/app/contact/contact.component.html:942]: Remove blank space

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html [hunk 6]
```diff
@@ -1,602 +1,536 @@
...
+                    <span (click)="modalRef.hide()"><img src="../../../assets/images/calender.svg"></span>
+                </div>
+                <div class="start-end-date">
+                    <div class="pos-rel mr-1">
+                        <label>Start Date</label>
+                        <input type="tel" name="inlineStartDate" [value]="viewOnlyStartDate" mask="00-00-0000"
+                            (blur)="saveInlineDates($event)">
+                        <span class="error-msg" *ngIf="invalidInlineStartDate">{{invalidInlineStartDate}}</span>
+                    </div>
+                    <div class="pos-rel">
```
Comment (sadik-walkover) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html:104]: use class name relative  everywhere

File: apps/web-giddh/src/assets/css/color.scss [hunk 7]
```diff
@@ -24,6 +24,8 @@ $cerulean: #0d83b3;
 $corn-flower-blue: #5092f3;
 $dark-midnight-blue: #040a58;
 $navy: #000962;
+$whiteSmoke: #f2f2f2;
```
Comment (sadik-walkover) [inline apps/web-giddh/src/assets/css/color.scss:27]: use color var name white-smoke

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 8]
```diff
@@ -8885,6 +8887,75 @@ input.checkbox-custom[type="checkbox"] {
...
+                padding: 0 5px;
+                top: -6px;
+                left: 9px;
+                font-size: 12px;
+                color: $gray;
+            }
+            input{
+                width: 130px;
+                height: 51px;
+                border: 1px solid #d9d9d9;
```
Comment (sadik-walkover) [inline apps/web-giddh/src/assets/css/style-2.scss:8942]: use var name everywhere

File: apps/web-giddh/src/app/contact/contact.component.html (hunk 1 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/contact/contact.component.html]: removed

File: apps/web-giddh/src/app/contact/contact.component.html (hunk 2 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/contact/contact.component.html]: updated

File: apps/web-giddh/src/app/contact/contact.component.html (hunk 3 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/contact/contact.component.html]: updated

File: apps/web-giddh/src/app/contact/contact.component.html (hunk 4 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/contact/contact.component.html]: updated

File: apps/web-giddh/src/app/contact/contact.component.html (hunk 5 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/contact/contact.component.html]: removed

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html (hunk 6 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html]: updated

File: apps/web-giddh/src/assets/css/color.scss (hunk 7 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/assets/css/color.scss]: changed

File: apps/web-giddh/src/assets/css/style-2.scss (hunk 8 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/assets/css/style-2.scss]: updated

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html [hunk 9]
```diff
@@ -1,602 +1,536 @@
...
+                </div>
+                <perfect-scrollbar fxFlex="auto" class="hide-horizontal-scroll" [scrollIndicators]="true"
+                    [config]="{suppressScrollX: false, suppressScrollY: false}">
+                    <div class="d-container" [style.max-height.px]="320" *ngIf="showMonthPicker">
+                        <div *ngFor="let year of allowedYears; let i = index;">
+                            <h2>
+                                {{ minDate.clone().add(i, 'year').format(" YYYY")}}
+                            </h2>
+                            <div class="dropdown-containt">
+                                <div class="d-month" (click)="setMonth(minDate.clone().add(i, 'year').year(),00)">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html:212]: Consider using an array to render this data to reduce the HTML size

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html [hunk 10]
```diff
@@ -1,602 +1,536 @@
...
-                        </option>
-                      </select>
+                    <!-- month name -->
+                    <!-- month selection -->
+
+                    <!-- month selection -->
+                </div>
+                <perfect-scrollbar fxFlex="auto" class="hide-horizontal-scroll" [scrollIndicators]="true"
+                    [config]="{suppressScrollX: false, suppressScrollY: false}">
+                    <div class="d-container" [style.max-height.px]="320" *ngIf="showMonthPicker">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html:206]: Avoid using inline styles

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html [hunk 11]
```diff
@@ -1,602 +1,536 @@
...
+                                    </select>
+                                    <select class="minuteselect" [disabled]="!endDate"
+                                        [(ngModel)]="timepickerVariables.start.selectedMinute"
+                                        (ngModelChange)="timeChanged($event, sideEnum.start)">
+                                        <option *ngFor="let i of timepickerVariables.start.minutes; let index = index;"
+                                            [value]="i"
+                                            [disabled]="timepickerVariables.start.disabledMinutes.indexOf(i) > -1">
+                                            {{timepickerVariables.start.minutesLabel[index]}}</option>
+                                    </select>
+                                    <select *ngIf="timePickerSeconds" class="secondselect" [disabled]="!endDate"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html:357]: Do use dash-case for class names

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html [hunk 12]
```diff
@@ -1,602 +1,536 @@
...
+                                </div>
+                                <div class="calendar-time" *ngIf="timePicker">
+                                    <select class="hourselect" [disabled]="!endDate"
+                                        [(ngModel)]="timepickerVariables.start.selectedHour"
+                                        (ngModelChange)="timeChanged($event, sideEnum.start)">
+                                        <option *ngFor="let i of timepickerVariables.start.hours" [value]="i"
+                                            [disabled]="timepickerVariables.start.disabledHours.indexOf(i) > -1">{{i}}
+                                        </option>
+                                    </select>
+                                    <select class="minuteselect" [disabled]="!endDate"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html:349]: Do use dash-case for class names

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html [hunk 13]
```diff
@@ -1,602 +1,536 @@
...
+                                                        <span
+                                                            *ngIf="(calendarMonth.start.calendar[1][1].month() == calendarMonth.start.calendar[row][col].month())">{{calendarMonth.start.calendar[row][col].date()}}</span>
+                                                    </td>
+                                                </tr>
+                                            </ng-container>
+                                        </tbody>
+                                    </table>
+                                </div>
+                                <div class="calendar-time" *ngIf="timePicker">
+                                    <select class="hourselect" [disabled]="!endDate"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.html:342]: Do use dash-case for class names

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 14]
```diff
@@ -574,68 +555,128 @@ $input-height: 3rem !default;
...
-            list-style: none;
-            margin: 0 auto;
-            padding: 0;
-            width: 193px;
+                        .drop-down {
+                            position: absolute;
+                            z-index: 1017;
+                            min-width: 96px;
+                            background-color: white;
+                            box-shadow: 0px 2px 8px #8e8383bf;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:592]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 15]
```diff
@@ -574,68 +555,128 @@ $input-height: 3rem !default;
...
+                            height: 23px;
+                            width: 23px;
+                            display: flex;
+                            background-size: 7px;
+                            background-position-x: center;
+                            background-position-y: center;
+                            border-radius: 50%;
+
+                            &:active {
+                                background-color: #afaaaabf;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:610]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 16]
```diff
@@ -574,68 +555,128 @@ $input-height: 3rem !default;
...
-                        z-index: 1017;
-                        min-width: 96px;
-                        background-color: white;
-                        box-shadow: 0px 2px 8px #8e8383bf;
-                        left: 28px;
-                        top: -3px;
-                        border-radius: 2px;
-                    }
+                        &.active {
+                            background-color: #e4e4e4 !important;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:615]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 17]
```diff
@@ -655,71 +696,62 @@ $input-height: 3rem !default;
...
-        &:hover, &:focus {
-            background-color: #3f51b5;
-        }
-
-        & > * {
-            position: relative;
+    .btn.btn-danger {
+        background: #e4e4e4;
+        color: #e51f0c;
+        border: 1px solide #e4e4e4;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:722]: Do use correct spelling
Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 18]
```diff
@@ -655,71 +696,62 @@ $input-height: 3rem !default;
...
+        margin-right: 10px;
+
+        &:hover,
+        &:active,
+        &:focus,
+        &:active:focus {
+            background: #d3d3d3;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:729]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 19]
```diff
@@ -655,71 +696,62 @@ $input-height: 3rem !default;
...
-            }
+    .btn.btn-success {
+        background: #0cb1af;
+        color: #fff;
+        border: 1px solid #0cb1af;
+        &:hover,
+        &:active,
+        &:focus,
+        &:active:focus {
+            background: #087e7d;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:748]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 20]
```diff
@@ -873,52 +940,61 @@ $input-height: 3rem !default;
...
     border-radius: 0 !important;
 }
-
+td span {
+    border: 1px solid transparent;
+}
 td:hover span {
-    background-color: #f8f8f8;
-    border: 1px solid;
+    background-color: #f2f2f2;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:961]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 21]
```diff
@@ -873,52 +940,61 @@ $input-height: 3rem !default;
...
 }
 
 .md-drppicker td.end-date.active {
     background-color: white !important;
 
+
     span {
-        background-color: #36469d;
+        background-color: #7d87e400 !important;
+        color: #1c1c1c !important;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:987]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 22]
```diff
@@ -873,52 +940,61 @@ $input-height: 3rem !default;
...
     }
 
     .available:hover {
         background-color: unset !important;
-
+    }
+    &:hover{
+        span{
+        background-color: #f2f2f2 !important;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:995]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 23]
```diff
@@ -931,53 +1007,52 @@ td span {
...
         text-align: center;
         text-transform: uppercase;
-        font-size: 13px;
+        font-size: 12px;
+        font-style: normal;
 
         &:hover {
             border-radius: 50px;
-            background-color: gainsboro;
+            background-color:#e4e4e4;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:1055]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 24]
```diff
@@ -987,60 +1062,59 @@ td span {
...
-}
-
 .activatedDate {
     background-color: #c6dafc;
 }
 
 .activatedDate span {
-    color: #000 !important;
-    background-color: #c6dafc !important;
+    color: #1c1c1c !important;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:1116]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 25]
```diff
@@ -1057,15 +1131,30 @@ td span {
 
 .sub-ranges-cal {
     position: absolute;
-    background-color: #f5f5f5;
+    background-color: #fff;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:1134]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 26]
```diff
@@ -1074,15 +1163,493 @@ td span {
...
+        display: flex;
+    }
+}
+.prev-next-btn {
+    width: 100px;
+    display: flex;
+    justify-content: space-around;
+}
+.days-upto-select {
+    border-bottom: 1px solid gainsboro;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:1286]: Incorrect use of SASS variables (missing $)

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 27]
```diff
@@ -1074,15 +1163,493 @@ td span {
...
+                cursor: pointer;
+                font-size: 12px;
+                
+
+                .drop-down {
+                    position: absolute;
+                    z-index: 1017;
+                    min-width: 96px;
+                    background-color: white;
+                    box-shadow: 0px 2px 8px #8e8383bf;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:1311]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 28]
```diff
@@ -1074,15 +1163,493 @@ td span {
...
+                    height: 23px;
+                    width: 23px;
+                    display: flex;
+                    background-size: 7px;
+                    background-position-x: center;
+                    background-position-y: center;
+                    border-radius: 50%;
+
+                    &:active {
+                        background-color: #afaaaabf;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:1329]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 29]
```diff
@@ -1074,15 +1163,493 @@ td span {
...
+                    }
+                }
+
+                &.active {
+                    background-color: #e4e4e4 !important;
+                    border-left: 3px solid #3f51b5 !important;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:1335]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 30]
```diff
@@ -1074,15 +1163,493 @@ td span {
...
+                width:100%;
+            }
+        }
+
+        .calender-container{
+            position: absolute;
+            top: 0;
+            width: 100%;
+            left: 0;
+            background-color: #fff;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:1478]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 31]
```diff
@@ -1074,15 +1163,493 @@ td span {
...
+    perfect-scrollbar{
+        max-height: 90% !important;
+        height: calc(100% - 100px) !important;
+    } 
+
+    .bottom-botton{
+        position: fixed;
+        bottom: 0;
+        right: 0;
+        background-color: #fff;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:1610]: Use variables

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.ts [hunk 32]
```diff
@@ -246,14 +290,75 @@ export class NgxDaterangepickerComponent implements OnInit {
                 this.locale.format = moment.localeData().longDateFormat('L');
             }
         }
-        this.renderCalendar(DateType.start);
-        this.renderCalendar(DateType.end);
+
+        this.initCalendar();
+
+        this.router.events.subscribe(event => {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.ts:296]: Do use takeUntil

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss (hunk 28 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss]: updated

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss (hunk 29 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss]: updated

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss (hunk 30 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss]: updated

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss (hunk 31 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss]: updated

File: apps/web-giddh/src/app/contact/contact.component.ts [hunk 33]
```diff
@@ -853,9 +848,11 @@ export class ContactComponent implements OnInit, OnDestroy, OnChanges {
     }
 
     public selectedDate(value: any) {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/contact/contact.component.ts:850]: add doc

File: apps/web-giddh/src/app/contact/contact.component.ts [hunk 34]
```diff
@@ -853,9 +848,11 @@ export class ContactComponent implements OnInit, OnDestroy, OnChanges {
...
-        this.fromDate = moment(value.picker.startDate).format('DD-MM-YYYY');
-        this.toDate = moment(value.picker.endDate).format('DD-MM-YYYY');
-        if (value.event.type === 'hide') {
+        if (value.startDate && value.endDate) {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/contact/contact.component.ts:851]: nullity check for value

File: apps/web-giddh/src/app/contact/contact.component.ts [hunk 35]
```diff
@@ -1179,4 +1176,19 @@ export class ContactComponent implements OnInit, OnDestroy, OnChanges {
...
     }
 
+    /**
+     * This will toggle all columns
+     *
+     * @param {boolean} event
+     * @memberof ContactComponent
+     */
+    public selectAllColumns(event: boolean): void {
+        this.showFieldFilter.parentGroup = event;
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/contact/contact.component.ts:1186]: nullity check for event

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.ts [hunk 36]
```diff
@@ -610,7 +695,13 @@ export class NgxDaterangepickerComponent implements OnInit {
         this.updateView();
     }
 
-    setEndDate(endDate) {
+    public setEndDate(endDate): void {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.ts:698]: nullity check for endDate

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 37]
```diff
@@ -574,68 +555,128 @@ $input-height: 3rem !default;
         float: none;
         text-align: left;
         margin: 0;
+        border-right: 1px solid gainsboro;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:558]: Incorrect usage of SASS variable

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss [hunk 38]
```diff
@@ -1074,15 +1163,493 @@ td span {
...
+}
+.off.in-range-hovered{
+    &::before{
+        display: none;
+    }
+}
+.in-range-hovered {
+        &::before {
+            content: "";
+            border-bottom: 1px dashed gray;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss:1379]: Consider using $ in color

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss (hunk 23 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss]: updated

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss (hunk 24 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss]: updated

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss (hunk 26 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss]: updated

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss (hunk 27 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss]: updated

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss (hunk 38 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss]: updated

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss (hunk 37 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.scss]: updated

File: apps/web-giddh/src/app/contact/contact.component.ts (hunk 33 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/contact/contact.component.ts:850]: It's old function.

File: apps/web-giddh/src/app/contact/contact.component.ts (hunk 34 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/contact/contact.component.ts:851]: solved

File: apps/web-giddh/src/app/contact/contact.component.ts (hunk 35 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/contact/contact.component.ts:1186]: Not required.

File: apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.ts (hunk 36 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/theme/ngx-date-range-picker/ngx-daterangepicker.component.ts:698]: Did you check the whole code? Do you think it's required?

**Pull request merged**